### PR TITLE
bugfix > fix behavior change by making fields not mandatory

### PR DESCRIPTION
**Pull request by DC2-DanielKrueger**
**Motivation**

Resolves #<issue>

**Changes**

### CODE REVIEW

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/config/impl/AbstractProtocolAdapterConfig.java [hunk 1]
```diff
@@ -69,22 +69,22 @@ public enum MessageHandlingOptions {
                            defaultValue = "0")
         private int qos = 0;
 
-        @JsonProperty(value = "messageHandlingOptions", required = true)
+        @JsonProperty(value = "messageHandlingOptions", required = false)
         @ModuleConfigField(title = "Message Handling Options",
                            description = "The topic to publish data on",
```
Comment (schaebo) [inline hivemq-edge/src/main/java/com/hivemq/edge/modules/config/impl/AbstractProtocolAdapterConfig.java:74]: not really your change, but just noticed tat the description is not correct here.

```suggestion
                           description = "This setting defines the format of the resulting MQTT message, either a message per changed tag or a message per subscription that may include multiple data points per sample",
```

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/config/impl/AbstractProtocolAdapterConfig.java [hunk 2]
```diff
@@ -69,22 +69,22 @@ public enum MessageHandlingOptions {
...
         @ModuleConfigField(title = "Include Sample Timestamp In Publish?",
                            description = "Include the unix timestamp of the sample time in the resulting MQTT message",
                            defaultValue = "true")
         private @Nullable Boolean includeTimestamp = Boolean.TRUE;
 
-        @JsonProperty(value = "includeTagNames", required = true)
+        @JsonProperty(value = "includeTagNames", required = false)
         @ModuleConfigField(title = "Include Tag Names In Publish?",
                            description = "Include the names of the tags in the resulting MQTT publish",
                            defaultValue = "true")
```
Comment (schaebo) [inline hivemq-edge/src/main/java/com/hivemq/edge/modules/config/impl/AbstractProtocolAdapterConfig.java:90]:
```suggestion
                           defaultValue = "false")
```

**Review by schaebo — APPROVED** [review]
a few small suggestions

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/config/impl/AbstractProtocolAdapterConfig.java [hunk 3]
```diff
@@ -69,25 +69,25 @@ public enum MessageHandlingOptions {
...
         private @Nullable Boolean includeTimestamp = Boolean.TRUE;
 
-        @JsonProperty(value = "includeTagNames", required = true)
+        @JsonProperty(value = "includeTagNames", required = false)
         @ModuleConfigField(title = "Include Tag Names In Publish?",
                            description = "Include the names of the tags in the resulting MQTT publish",
-                           defaultValue = "true")
+                           defaultValue = "false")
         private @Nullable Boolean includeTagNames = Boolean.TRUE;
```
Comment (DC2-DanielKrueger) [inline hivemq-edge/src/main/java/com/hivemq/edge/modules/config/impl/AbstractProtocolAdapterConfig.java:91]:
```suggestion
        private @Nullable Boolean includeTagNames = Boolean.FALSE;
```

**Pull request merged**